src/http_request.rs

```diff
@@ -37,27 +41,41 @@ impl HttpRequest {
         // If the request is complete we are returning the response in the stream
         if request_status.is_complete() {
             info!("Request is complete.");
-            match request.path {
-                Some(ref _path) => {
-                    info!("Request path is: {}", _path);
-                    let response = "HTTP/1.1 200 OK\r\n\r\n";
-                    stream.write_all(response.as_bytes()).unwrap();
-                    stream.flush().unwrap();
-                }
-                None => {
-                    info!("Request path is missing.");
-                }
-            }
+            let path = request.path.map(String::from);
+            let method = match request.method.unwrap() {
+                "POST" => Some(MethodType::POST),
+                "GET" => Some(MethodType::GET),
+                "PATCH" => Some(MethodType::PATCH),
+                "DELETE" => Some(MethodType::DELETE),
+                "CONNECT" => Some(MethodType::CONNECT),
+                "OPTIONS" => Some(MethodType::OPTIONS),
+                "TRACE" => Some(MethodType::TRACE),
+                "HEAD" => Some(MethodType::HEAD),
+                _ => None,
+            };
+            Some(HttpRequest { path, method })
         } else {
             // TODO: Determine what to do if request is not complete
             info!("Request is partial.");
+            None
         }
     }
+
+    pub fn path(&self) -> &Option<String> {
+        &self.path
+    }
+
+    pub fn method(&self) -> &Option<MethodType> {
+        &self.method
```

Comment:
self.method.as_ref() ?

Reply:
The first was pretty easy, this one is a bit harder, I don't know how to impl the trait `AsRef` on an enum.

Reply:
Ok, so I modify the structure, it's no more an Option, I will return directly `&MethodType`. 
I removed the optionals on the struct, because it must not be optional, but mandatory :)

Reply:
Imo, it's not necessary to pass the result by the reference: using enums here is quite cheap and can be easily retured by value

Reply:
A method isnt optional indeed, good call!